src/stores/mainStore.ts

```diff
@@ -55,13 +68,37 @@ export class MainStore {
     );
     this.game.setWinner(winner);
     if (winner === WinnerOptions.Player) {
-      this.player.addPlayerMoney(this.game.playerBet);
+      this.player.addPlayerMoney(this.game.playerBet * 2);
+      // this.player.removePlayerMoney(this.game.bankerBet + this.game.tieBet);
     }
     if (winner === WinnerOptions.Tie) {
-      this.player.addPlayerMoney(this.game.tieBet * 5);
+      this.player.addPlayerMoney(this.game.tieBet * 6);
+      // this.player.removePlayerMoney(this.game.bankerBet + this.game.playerBet);
     }
     if (winner === WinnerOptions.Banker) {
-      this.player.addPlayerMoney(this.game.bankerBet * 0.95);
+      this.player.addPlayerMoney(this.game.bankerBet * 1.95);
+      // this.player.removePlayerMoney(this.game.playerBet + this.game.tieBet);
+    }
+  }
+
+  createSnapshot() {
+    const snapshot = new Snapshot(
+      this.game,
+      this.player,
+      this.game.playerBet,
+      this.game.tieBet,
+      this.game.bankerBet,
+      this.player.playerMoney
+    );
+
+    this.snapshots.push(snapshot);
+  }
+
+  undoLastBet() {
+    if (this.snapshots.length > 1) {
+      const snapshotToRestore = this.snapshots.at(-2);
```

Comment:
**.at** nice method but compatibility is not that nice. Once I got a bug because of Safari 😅

But you can use it in your project. This comment is more like **FYI**. Just keep in mind that some times you'd want to check documentation Browser compatibility [section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility)